langserver/handlers/did_change.go

```diff
@@ -64,13 +64,17 @@ func TextDocumentDidChange(ctx context.Context, params DidChangeTextDocumentPara
 	if err != nil {
 		return err
 	}
+
 	rms := cf.RootModuleCandidatesByPath(fh.Dir())
 	if len(rms) > 0 {
+		// Here we assume the first match is exact match
+		// TODO: Verify *here* this is in fact the case
 		rm := rms[0]
-		err := rm.ParseProviderReferences()
+		err := rm.ParseAndLoadFiles()
 		if err != nil {
 			return err
 		}
+		// TODO: publish/update diags from rm.ParsedDiagnostics()
```

Comment:
hmm I guess this changes the diagnostics implementation, fortunately it's not many lines.